lib/src/modal/modal.component.ts

```diff
@@ -0,0 +1,102 @@
+})
+export class ModalComponent {
+    @Input() id?: string;
+    @Input() size?: ModalSizeType;
+    @Input() center?: boolean;
+    @Input() position: ModalPositionType;
+    @Input() fullscreen?: boolean;
+    @Input() backdropDismiss?: boolean = true;
+    @Input() escapeKeyDismiss?: boolean = true;
+    @Input() className?: string;
+    @Input() ariaLabel?: string;
+    @Input() ariaDescribedby?: string;
+    @Input() animationDuration?: string = ".15s";
+    @ViewChild("modalRef") modalRef: ElementRef;
+    backDropRef: ComponentRef<SebModalBackdropComponent>;
+    toggle: boolean; // toggle is required to enable the open or close animation
+
+    constructor(private modalService: ModalService) {}
+
+    /**
+     * construct the class names that needs to be appended to the modal depending on the inputs requested
+     * @returns { [key: string]: boolean } key value pair for the positions of the modal
+     */
+    get modalPosition(): { [key: string]: boolean } {
+        return {
+            "modal-aside": !!this.position && !this.fullscreen,
+            "modal-aside-left": this.position === "left",
+            "modal-aside-right": this.position === "right",
+            "modal-fullscreen": this.fullscreen,
+        };
+    }
+
+    /**
+     * emit close event when backdrop is clicked
+     */
+    @HostListener("click", ["$event"])
+    onBackdropClick(event: MouseEvent): void {
+        if (this.backdropDismiss) {
+            if (event && event.target && event.target["classList"] && event.target["classList"].length) {
+                const classList: DOMTokenList = event.target["classList"];
+                if (classList.contains("modal")) {
+                    this.close();
+                }
+            }
+        }
+    }
+
+    /**
+     * emit close when escape key is clicked
+     */
+    @HostListener("keyup", ["$event"])
+    onEscKey(event: KeyboardEvent): void {
+        if (this.escapeKeyDismiss && event.keyCode === 27) {
+            this.close();
+        }
+    }
+
+    get toggleState(): string {
+        return this.toggle ? "open" : "close";
+    }
+
+    /**
+     * append backrop to the body, open the modal and trigger the animation
+     */
+    open(): void {
+        this.backDropRef = this.modalService.appendComponentToBody(SebModalBackdropComponent);
+        this.modalService.open(this.modalRef);
+        this.toggle = true;
+    }
+
+    /**
+     * close the modal, trigger the anumation and remove the backdrop from the body using its reference
+     */
+    close(): void {
+        this.modalService.close(this.modalRef);
+        this.toggle = false;
+        this.modalService.removeComponentFromBody(this.backDropRef);
+    }
+
+    /**
+     * callback when the animation is done
+     * @param event
+     */
+    onAnimationDone(event: AnimationEvent): void {
+        if (event.toState === "open") {
+            setTimeout(() => this.modalRef.nativeElement.focus(), 0);
```

Comment:
you could wrap this setTimeout in `runOutsideAngular` to avoid retriggering change detection as mentioned here
https://stackoverflow.com/questions/35752237/angular-2-focus-doesnt-work-if-i-dont-use-timeout

Reply:
This method is a callback for when the open animation finishes, this won't trigger a change detection, since nothing changed.
In fact this function only runs after the change happened.